Recommend diff scans and full scans on separate branches (#357)
returntocorp/semgrep-docs: docs/semgrep-ci/overview.md

[Semgrep CI](https://github.com/returntocorp/semgrep-action) (aka Semgrep Action or `semgrep-agent`) is a specialized Docker image for running Semgrep in CI environments. It can also optionally connect to [Semgrep App](https://semgrep.dev/manage) for centralized rule and findings management.

> - **Scan every commit**. Semgrep CI rapidly scans modified files on pull and merge requests, protecting developer productivity. In most successful configurations, full project scans are reserved for specific branches, such as trunk branches, and diff-aware scans are done on other branches before merging them into the trunk.

Reviewer: ```suggestion
- **Scan every commit**. Semgrep CI rapidly scans modified files on pull and merge requests, protecting developer productivity. Usually full project scans are reserved for specific branches, such as trunk branches, and diff-aware scans are done on other branches before merging them into the trunk.
```

Author: I didn't love this either... tried a third option 😄
